scripts/release.js

```diff
@@ -1,211 +1,168 @@
+
+  process.chdir(path.join(__dirname, '../../amazon-chime-sdk/apps/meeting/serverless'));
+  const formattedVersion = version.replace(/\./g, '-');
+  spawnOrFail('node', [`./deploy.js -r us-east-1 -b chime-sdk-components-demo-${formattedVersion} -s chime-sdk-components-demo-${formattedVersion}`], { printErr: true });
+
+  process.chdir(path.join(__dirname, '..'));
+  spawnOrFail('rm', ['-rf ../amazon-chime-sdk']);
+};
+
+const getCurrentRemoteBranch = () => {
+  return (spawnOrFail('git', ['for-each-ref --format="%(upstream:short)" "$(git symbolic-ref -q HEAD)"'], { skipOutput: true })).trim();
+};
+
+const buildAndPack = () => {
+  logger.log('Building package...');
+  spawnOrFail('npm', [`run build:publish`]);
+  logger.log('Packaging ...');
+  spawnOrFail('npm', ['pack'], { printErr: true });
+};
+
+const updateJSSdkDep = () => {
   // Update the peer dependency to the most updated version of the SDK
-  logger.log(`Installing SDK Version: ${updatedSdkVersion} as a peerDependency and devDependency of the react library.`);
+  logger.log(`Installing SDK Version: ${latestNPMJSSdkVersion} as a peerDependency and devDependency of the react library.`);
 
   let componentsPackageJson = JSON.parse(fs.readFileSync('package.json', 'utf-8'));
-  componentsPackageJson.peerDependencies['amazon-chime-sdk-js'] = `^${updatedSdkVersion}`;
-  componentsPackageJson.devDependencies['amazon-chime-sdk-js'] = `^${updatedSdkVersion}`;
+  componentsPackageJson.peerDependencies['amazon-chime-sdk-js'] = `^${latestNPMJSSdkVersion}`;
+  componentsPackageJson.devDependencies['amazon-chime-sdk-js'] = `^${latestNPMJSSdkVersion}`;
 
   fs.writeFileSync(
     'package.json',
     JSON.stringify(componentsPackageJson, null, 2)
   );
 
-  logger.log("NPM Installing component library...")
   spawnOrFail('npm', [`install`]);
-
+};
+
+const checkNPMDepsInstall = () => {
+  // Creates a new React test app, install component and sdk and build the test app to ensure no peer dependency warnings, errors or build issues
+  logger.log('Install amazon-chime-sdk-component-library-react and amazon-chime-sdk-js sdk as dependencies, check if there is peer dependency warning for amazon-chime-sdk-component-library-react');
+  process.chdir(path.join(__dirname, '../..'));
+  if (!fs.existsSync('dependency-check-app')){
+    spawnOrFail('mkdir', ['dependency-check-app']);
+  } else {
+    spawnOrFail('rm', ['-rf dependency-check-app']);
+    spawnOrFail('mkdir', ['dependency-check-app']);
+  }
+  process.chdir(path.join(__dirname, '../../dependency-check-app'));
+  spawnOrFail('npm', ['init -y']);
+  spawnOrFail('npm', ['install react react-dom']);
+  spawnOrFail('npm', [`install amazon-chime-sdk-js@${latestNPMJSSdkVersion} styled-components styled-system`]);
+  checkWarning('npm', [`install -q ../amazon-chime-sdk-component-library-react/amazon-chime-sdk-component-library-react-${currentVersion}.tgz`], null, 'amazon-chime-sdk-component-library-react');
```

Comment:
What is this `-q` here?

Reply:
This is from the existing release script so not sure. I think this is what Lu/Jing added to check on dependency installation. However, I don't think it still work (I tried it out with npm 8 and it just auto resolves the conflict by updating js sdk version). Guess it is out of scope of this so we can come back to this later.